session-manager/src/session_context.rs

```diff
@@ -145,20 +152,29 @@ impl SessionContext {
             let cert: rustls::Certificate = rustls::Certificate(this_chain_data.clone());
             cert_chain.push(cert);
         }
-        self.server_config.set_single_cert(cert_chain, self.server_private_key.clone())?;
+        match &mut self.server_config {
+            Some(config) => config.set_single_cert(cert_chain, self.server_private_key.clone())?,
+            None => return Err(SessionManagerError::InvalidStateError),
+        }
         return Ok(());
     }
 
     /// Returns the configuration associated with the server.
     #[inline]
-    pub fn server_config(&self) -> &ServerConfig {
-        &self.server_config
+    pub fn server_config(&self) -> Result<ServerConfig, SessionManagerError> {
```

Comment:
Similarly, should this not return a borrow of `ServerConfig` in the `Ok(..)` case?

Reply:
Changed